File: booking-kit/src/main/java/com/skedgo/android/bookingkit/model/StepperFormField.java
package com.skedgo.android.bookingkit.model;

import android.os.Parcel;

import com.google.gson.annotations.SerializedName;

public class StepperFormField extends FormField {
  public static final Creator<StepperFormField> CREATOR = new Creator<StepperFormField>() {
    @Override
    public StepperFormField createFromParcel(Parcel in) {
      in.readInt();
      return new StepperFormField(in);
    }

    @Override
    public StepperFormField[] newArray(int size) {
      return new StepperFormField[size];
    }
  };
  @SerializedName("value")
  private int value;
  @SerializedName("minValue")
  private int minValue;
  @SerializedName("maxValue")
  private int maxValue;

  public StepperFormField() {
    super();
  }

  public StepperFormField(Parcel in) {
    super(in);
    this.value = in.readInt();
    this.minValue = in.readInt();
    this.maxValue = in.readInt();
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeInt(STEPPER);
    super.writeToParcel(dest, flags);
    dest.writeInt(value);
    dest.writeInt(minValue);
    dest.writeInt(maxValue);
  }

  @Override
  public Integer getValue() {
    return value;
  }

  public void setValue(int value) {
    this.value = value;
  }

  public int getMinValue() {
    return minValue;
  }

  public void setMinValue(int minValue) {
    this.minValue = minValue;
  }

  public int getMaxValue() {
    return maxValue;
  }

  public void setMaxValue(int maxValue) {
    this.maxValue = maxValue;
  }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/StepperFormField.kt
package com.skedgo.tripkit.booking

import android.os.Parcel
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

class StepperFormField : FormField {

    @SerializedName("value")
    var mValue: Int = 0

    @SerializedName("minValue")
    var minValue: Int = 0

    @SerializedName("maxValue")
    var maxValue: Int = 0

    constructor(parcel: Parcel) : super( parcel) {  // Call the parent class constructor
        mValue = parcel.readInt()
        minValue = parcel.readInt()
        maxValue = parcel.readInt()
    }

    // Secondary empty constructor (the default constructor)
    constructor() : super() {
        // Initialize any default values or leave it empty
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(STEPPER)
        super.writeToParcel(dest, flags)
        dest.writeInt(mValue)
        dest.writeInt(minValue)
        dest.writeInt(maxValue)
    }

    override fun getValue(): Int {
        return mValue
    }

    fun setValue(value: Int) {
        this.mValue = value
    }

    companion object CREATOR : Creator<StepperFormField> {
        override fun createFromParcel(parcel: Parcel): StepperFormField {
            parcel.readInt() // Read STEPPER type
            return StepperFormField(parcel)
        }

        override fun newArray(size: Int): Array<StepperFormField?> {
            return arrayOfNulls(size)
        }
    }
}